### PR TITLE
MESH-1824: Empty perms default + MESH-1825: Contracts RPC

### PR DESCRIPTION
**Pull request by Centril**
### modified API

- Contracts `instantiate` APIs now use empty permissions by default.
- Enable RPC for contracts.

### CODE REVIEW

File: pallets/runtime/common/src/runtime.rs [hunk 1]
```diff
@@ -725,34 +725,41 @@ macro_rules! runtime_apis {
...
+
+                fn instantiate(
+                    origin: polymesh_primitives::AccountId,
+                    endowment: Balance,
+                    gas_limit: u64,
+                    code: pallet_contracts_primitives::Code<polymesh_primitives::Hash>,
+                    data: Vec<u8>,
+                    salt: Vec<u8>,
+                ) -> pallet_contracts_primitives::ContractInstantiateResult<polymesh_primitives::AccountId> {
+                    Contracts::bare_instantiate(origin, endowment, gas_limit, code, data, salt, false)
```
Comment (Neopallium) [inline pallets/runtime/common/src/runtime.rs:753]: I think this RPC endpoint needs the logic from our contract pallet's `instantiate_with_code`, (i.e. attach the contract address to the caller's identity).

Comment (Neopallium) [reply pallets/runtime/common/src/runtime.rs:753]: I think these RPC endpoints are only for testing if an instantiate/call will work (i.e. dry-run).  It should be fine to use the `instantiate` call without permissions.  We just want this interface to be as compatible with the Contracts UI as possible.

**Review by CJP10 — APPROVED** [review]

File: pallets/contracts/src/lib.rs [hunk 2]
```diff
@@ -427,6 +494,50 @@ impl<T: Config> Module<T> {
...
+        salt: Vec<u8>,
+    ) -> ContractInstantiateResult<T::AccountId> {
+        match (|| {
+            // Ensure we have perms + we'll need DID.
+            let did =
+                pallet_permissions::Module::<T>::ensure_call_permissions(&sender)?.primary_did;
+
+            // Roll back `prepare_instantiate` if contract was not instantiated.
+            let code_hash = Self::code_hash(&code);
+            Self::prepare_instantiate(did, &sender, &code_hash, &salt, Permissions::empty())?;
```
Comment (Neopallium) [inline pallets/contracts/src/lib.rs:515]: There is no transaction here.  No `Roll back`.

**Review by Neopallium — CHANGES_REQUESTED** [review]
Looks good.  Just a minor comment change.

**Review by adamdossa — APPROVED** [review]

**Pull request merged**